staging/src/k8s.io/client-go/examples/read-yaml-configuration-deployment/main.go

```diff
@@ -0,0 +1,214 @@
+	flag.Parse()
+
+	// Get filename
+	yamlFilepath, err := filepath.Abs(*yamlfilename)
+
+	// Get reader from file opening
+	reader, err := os.Open(yamlFilepath)
+	if err != nil {
+		panic(err)
+	}
+
+	// Split YAML into chunks or k8s resources, respectively
+	yamlDecoder := yaml.NewDocumentDecoder(ioutil.NopCloser(reader))
+
+	// Create decoding function used for YAML to JSON decoding
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+
+	// Read first resource - expecting deployment with size < 2048
+	yamlDeployment := make([]byte, 2048)
+	_, err = yamlDecoder.Read(yamlDeployment)
+	if err != nil {
+		panic(err)
+	}
+
+	// Trim unnecessary trailing 0x0 signs which are not accepted
+	TrimmedYamlDeployment := strings.TrimRight(string(yamlDeployment), string(byte(0)))
+
+	// Decode deployment resource from YAML to JSON
+	jsonDeployment, _, err := decode([]byte(TrimmedYamlDeployment), nil, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Check "kind: deployment"
+	if jsonDeployment.GetObjectKind().GroupVersionKind().Kind != "Deployment" {
+		panic(fmt.Sprintf("\nError: Not \"Kind: Deployment\"\n%#v\n", jsonDeployment))
+	}
+
+	// Marshall JSON deployment
+	d, err := json.Marshal(&jsonDeployment)
+	if err != nil {
+		panic(err)
+	}
+
+	// Unmarshall JSON into deployment struct
+	var deployment appsv1beta1.Deployment
+	err = json.Unmarshal(d, &deployment)
+	if err != nil {
+		panic(err)
+	}
+
+	// Read second resource - expecting service with size < 1024
+	yamlService := make([]byte, 1024)
+	_, err = yamlDecoder.Read(yamlService)
+	if err != nil {
+		panic(err)
+	}
+
+	// Trim unnecessary trailing 0x0 signs which are not accepted
+	TrimmedYamlService := strings.TrimRight(string(yamlService), string(byte(0)))
+
+	// Decode service resource from YAML to JSON
+	jsonService, _, err := decode([]byte(TrimmedYamlService), nil, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// Check "kind: service"
+	if jsonService.GetObjectKind().GroupVersionKind().Kind != "Service" {
+		panic(fmt.Sprintf("Error: Not \"Kind: Service\"\n%#v\n", jsonService))
+	}
+
+	// Marshall JSON Service
+	s, err := json.Marshal(&jsonService)
+	if err != nil {
+		panic(err)
+	}
+
+	//Unmarshall JSON into service struct
+	var service corev1.Service
+	err = json.Unmarshal(s, &service)
+	if err != nil {
+		panic(err)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create clientset from outside of the cluster
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create client for deployments
+	deploymentsClient := clientset.AppsV1beta1().Deployments(corev1.NamespaceDefault)
+
+	// Create client for services
+	servicesClient := clientset.CoreV1().Services(corev1.NamespaceDefault)
+
+	// create deployment - try update on error
+	fmt.Printf("Create deployment %q\n", deployment.Name)
+	createdDeployment, err := deploymentsClient.Create(&deployment)
+	if err != nil {
```

Comment:
this should check the error. Only update if it existed already.

Reply:
I think I cannot reliably match the error string to figure out the 'is already running' error. Instead, I would try to check it out in advance.

Reply:
there are helpers like `k8s.io/apimachinery/pkg/api/errors.IsNotFound`